### PR TITLE
BSPIMX8M-3424 bsp: imx8mp: Add chapter about booting efi

### PR DESCRIPTION
**Pull request by BHahn42618**
For the new PD24.1.0 release booting with efi is supported. Add a chapter on how to turn on/off booting with efi and how to install a distro and boot it with efi.

### CODE REVIEW

File: source/bsp/efi.rsti [hunk 1]
```diff
@@ -0,0 +1,73 @@
+EFI Boot
+--------
+
+Standardboot in U-Boot also supports booting distros over efi. By default the U-Boot will search
+for a bootscript first, which is used to boot our Ampliphy distro. If it does not find any
```
Comment (ymoog) [inline source/bsp/efi.rsti:5]: This is imx8 specific behaviour, remove this please.

File: source/bsp/efi.rsti [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+bootscript, it will search for bootable efi applications. So for booting over efi just make sure
+you don't have our distro installed.
+
+Disable booting with efi
+........................
+
+To disable booting with efi you have to set the ``doefiboot`` variable to 0. Also make sure you
```
Comment (ymoog) [inline source/bsp/efi.rsti:12]: This is also imx8 specific, right?

File: source/bsp/efi.rsti [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+.. code-block:: console
+
+   sudo dd if=file.iso of=/dev/sdx bs=1M conv=fsync status=progress
+
+Insert the USB stick into the board and boot it. GRUB will then prompt you with a menu where you
+can select what to do. Here select install. Then you have to click through the installation menu.
+This is relatively straightforward and differs a bit for every distro. You can install the distro
+for example to emmc (mmc2) or sdcard (mmc1). Make sure you do not overwrite your U-Boot during the
+install.
```
Comment (ymoog) [inline source/bsp/efi.rsti:49]: Please describe in more detail what not to do. When will I overwrite the bootloader?

File: source/bsp/efi.rsti [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+(USB-stick). The board should then boot the distro you installed.
+
+If that does not happen, check if there is a boot option set for the distro. The easiest way is
+with the ``eficonfig`` command.
+
+.. code-block:: console
+
+   u-boot=> eficonfig
+
+That will open a menu. When you can select ``Edit Boot Option``. It will show you the current boot
```
Comment (ymoog) [inline source/bsp/efi.rsti:66]:
```suggestion
That will open a menu. Then you can select ``Edit Boot Option``. It will show you the current boot
```

**Review by ymoog — CHANGES_REQUESTED** [review]

File: source/bsp/efi.rsti (hunk 1 above)
Comment (BHahn42618) [reply source/bsp/efi.rsti]: Moved the imx8 specific part to imx8 folder

File: source/bsp/efi.rsti (hunk 2 above)
Comment (BHahn42618) [reply source/bsp/efi.rsti]: Moved the imx8 specific part to imx8 folder

File: source/bsp/efi.rsti (hunk 3 above)
Comment (BHahn42618) [reply source/bsp/efi.rsti]: Hope this is okay now

**Review by ymoog — APPROVED** [review]

File: source/bsp/efi.rsti (hunk 3 above)
Comment (ymoog) [reply source/bsp/efi.rsti]: Yes, looks good! Thanks

File: source/bsp/installing-distro-efi.rsti [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+Then copy the .iso file to a usb stick for example. Make sure you select the correct device:
+
+.. code-block:: console
+
+   sudo dd if=file.iso of=/dev/sdx bs=1M conv=fsync status=progress
+
+Insert the USB stick into the board and boot it. GRUB will then prompt you with a menu where you
+can select what to do. Here select install. Then you have to click through the installation menu.
+This is relatively straightforward and differs a bit for every distro. You can install the distro
+for example to emmc (mmc2) or sdcard (mmc1). Make sure you do not overwrite your U-Boot during the
```
Comment (tremmet) [inline source/bsp/installing-distro-efi.rsti:18]: What about using substitutions for mmc2 and mmc1? So it makes it a little bit more generic and more ready to use for different platform.

**Review by jonas-rem — APPROVED** [review]
Waiting for one more review from the imx8 team to check contents before merging.

**Review by tremmet — APPROVED** [review]

**Review by ymoog — APPROVED** [review]

**Pull request merged**